### DISCUSSION TITLE
Wireguard connection status

### DISCUSSION THREAD

**Discussion by Neurology0443** [celzero/rethink-app#1024] [Q&A]
First of all, thank you for your amazing work and time spend to make such a great app ! I'm brand new to your piece of software and It's looks amazing, feels amazing and is very useful for privacy & anonimity.

I use wireguard regularly and took me a while to make "it work" in rethink because I was waiting for it to send me some data/handshake about the connection. Like on the android, mac, win app, you can see the connection/handshake is complete based on "data recieved", "data send", which apear only when the handshake is completed and working.

Having a visual confirmation on a working connection (while not necessary), would assure the user that everything is working accordingly.

I have no idea on how to implemented it (not a progammer at all...) so it's more a suggestion than an issue and probably not that important?

Also, why aren't all apps selected by default in the wireguard connection? Isn't the purpose of a secure tunnel to pass everything through the encrypted tunnel?

Thanks guys !

**Comment by ignoramous** [comment]
Thanks for you kind words (:

> I was waiting for it to send me some data/handshake about the connection. Like on the android, mac, win app, you can see the connection/handshake is complete based on "data recieved", "data send", which apear only when the handshake is completed and working.
>
> Having a visual confirmation on a working connection (while not necessary), would assure the user that everything is working accordingly.

The app must show either `failing` or `connecting` against each WireGuard profile. Is that not the case? That said, I can see how the data indicators might help, will add those in the upcoming version: https://github.com/celzero/rethink-app/issues/1039

> Also, why aren't all apps selected by default in the wireguard connection? Isn't the purpose of a secure tunnel to pass everything through the encrypted tunnel?

That's because Rethink supports running multiple WireGuard profiles at once. However, we are impl a feature for apps not part of any *active* WireGuard profile to be tunneled to a *base* WireGuard profile: #1014 / also see: #976
